src/main/java/uk/gov/ch/developer/docs/interceptor/UserDetailsInterceptor.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.ch.developer.docs.interceptor;
+
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.lang.Nullable;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+import org.springframework.web.servlet.view.UrlBasedViewResolver;
+import uk.gov.ch.developer.docs.session.SessionService;
+
+@Component
+public class UserDetailsInterceptor extends HandlerInterceptorAdapter {
+
+    private static final String USER_EMAIL = "userEmail";
+    private static final String SIGN_IN_KEY = "signin_info";
+    private static final String USER_PROFILE_KEY = "user_profile";
+    private static final String EMAIL_KEY = "email";
+
+    @Autowired
+    private SessionService sessionService;
+
+    @Override
+    public void postHandle(HttpServletRequest request, HttpServletResponse response, Object handler,
+            @Nullable ModelAndView modelAndView) throws Exception {
+
+        if (modelAndView != null && modelAndView.getViewName() != null
+                && (request.getMethod().equalsIgnoreCase("GET")
+                || (request.getMethod().equalsIgnoreCase("POST")
+                && !modelAndView.getViewName()
+                .startsWith(UrlBasedViewResolver.REDIRECT_URL_PREFIX)))) {
+
+            Map<String, Object> sessionData = sessionService.getSessionDataFromContext();
```

Comment:
When me and David designed the User Model we extended it from LFPWeb. I think we were expecting the interceptor to have a (an autowired?) UserModel field and that would be populated running the populateUserDetails method against the user and then adding that to the model.
This way we don't have to change the User Interceptor if we need to update the user model